src/main/java/uk/gov/register/core/Item.java

```diff
@@ -73,14 +72,15 @@ public boolean equals(Object o) {
 
         Item item = (Item) o;
 
-        if (sha256hex != null ? !sha256hex.equals(item.sha256hex) : item.sha256hex != null) return false;
+        if (hashValue != null ? !hashValue.equals(item.hashValue) : item.hashValue != null) return false;
         return content != null ? content.equals(item.content) : item.content == null;
 
     }
 
     @Override
     public int hashCode() {
-        int result = sha256hex != null ? sha256hex.hashCode() : 0;
+        String hash = hashValue.getValue();
```

Comment:
You should not use `hashValue.getValue().getHashCode()` since you have already implemented `hashValue.getHashCode()`. You should just need to use `hashValue.getHashCode()`.